includes/event/event-repository.php

```diff
@@ -327,6 +327,51 @@ public function get_events_created_by_user( int $user_id, int $page = -1, int $p
 		// phpcs:enable
 	}
 
+	public function get_events_user_is_or_will_attend( int $user_id, int $page = -1, int $page_size = -1 ): Events_Query_Result {
```

Comment:
I think the existing `get_current_and_upcoming_events_for_user()` does the same as this method.